pulley/CONTRIBUTING.md

```diff
@@ -0,0 +1,233 @@
+
+Running our test again we get the same error as before! That leads us to the
+next part...
+
+#### Adding a Cranelift Lowering
+
+Next up we need to actually fix the error at hand, a new lowering rule needs to
+be added to Cranelift. Here we'll be working in
+`cranelift/codegen/src/isa/pulley_shared/lower.isle`. Cranelift instructions and
+ISLE rules for our new `xdiv32_s` instruction are automatically generated from
+the `for_each_op!` macro, so all we need to do is to add a new lowering rule.
+That'll look like so:
+
+```
+;;;; Rules for `idiv` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type $I32 (sdiv a b)))
+      (pulley_xdiv32_s a b))
+```
+
+Here the `lower` "rule" is what we're adding and is the main function used to
+lower from CLIF to Pulley bytecode. THe `pulley_xdiv32_s` constructor was
+auto-generated for us as part of `cranelift/codegen/meta`'s build.
+
+Running our test again yields:
+
+```
+Error: failed to run script file './tests/misc_testsuite/control-flow.wast'
+
+Caused by:
+    0: failed directive on ./tests/misc_testsuite/control-flow.wast:83:1
+    1: Compilation error: Unsupported feature: should be implemented in ISLE: inst = `v26 = band.i32 v2, v13  ; v13 = 3`, type = `Some(types::I32)`
+```
+
+Progress! This is a different error than before. Now it's time to rinse and
+repeat these steps. Be sure to skim the rest of `lower.isle` for inspiration on
+how to implement lowering rules. You can also look at `lower.isle` for other
+architecture backends too for inspiration.
+
+#### Flagging a test as passing
+
+After implementing a lowering for `band.i32` our test case is now passing:
+
+```
+$ cargo run --features pulley wast --target pulley64 ./tests/misc_testsuite/control-flow.wast
+    Finished `dev` profile [unoptimized + debuginfo] target(s) in 13.50s
+     Running `target/debug/wasmtime wast --target pulley64 ./tests/misc_testsuite/control-flow.wast`
+```
+
+If we run the test suite though we'll see:
+
+```
+$ cargo test --test wast control-flow.wast
+    Finished `test` profile [unoptimized + debuginfo] target(s) in 29.14s
+     Running tests/wast.rs (target/debug/deps/wast-f83a3ee5e5dbacde)
+
+running 6 tests
+...F.F
+failures:
+
+---- CraneliftPulley/./tests/misc_testsuite/control-flow.wast ----
+this test is flagged as should-fail but it succeeded
+
+---- CraneliftPulley/pooling/./tests/misc_testsuite/control-flow.wast ----
+this test is flagged as should-fail but it succeeded
+
+
+failures:
+    CraneliftPulley/./tests/misc_testsuite/control-flow.wast
+    CraneliftPulley/pooling/./tests/misc_testsuite/control-flow.wast
+
+test result: FAILED. 4 passed; 2 failed; 0 ignored; 0 measured; 4086 filtered out; finished in 0.05s
+
+error: test failed, to rerun pass `--test wast`
+```
+
+This indicates that the test was previously flagged as "should fail", but that
+assertion is no longer true! Update `crates/wast-util/src/lib.rs` about the new
+test passing and we'll see:
```

Comment:
```suggestion
assertion is no longer true! Update the `WastTest::should_fail` method in `crates/wast-util/src/lib.rs` so that it expects the test to pass and we'll see:
```